backend/controllers/user.controller.js

```diff
@@ -134,4 +101,121 @@ export const signin = async(req, res, next) => {  // email or username
         }catch(err) {
             next(err);
         }
+};
+
+export const getUsers = async(req, res, next) => {
+    try {
+        const { page = 1, limit = 25 } = req.query;
+        const users = await User.find().select('-password')
+        .limit(parseInt(limit))
+        .skip((page - 1) * limit);
+
+        // Get total count of users for pagination info
+        const count = await User.countDocuments();
+
+        res.status(200).json({
+          success: true,
+          total: count,
+          page: parseInt(page),
+          limit: parseInt(limit),
+          data: users
+        });
+      } catch (error) {
+         console.log(error);
+      }
+};
+
+export const getUserByAccountNumber = async (req, res, next) => {
+    const { id } = req.params;
+    if (!id) {
+        return next(errorHandler(400, 'Account Number is required'));
+    }
+    
+    try {
+        const validUser = await User.findOne({accountNumber : id}).select("-password");
+
+        if (!validUser) {
+            return next(errorHandler(404, 'User not found'));
+        }
+        
+        res.status(200).json({
+            success: true,
+            data : validUser
+        })
+    }catch(error) {
+       next(error);
+    }
+};
+
+export const updateUser = async(req, res, next) => {
+    const modifyUser = req.body;
+    const { id } = req.params;
+    if (!id) {
+        return next(errorHandler(400, 'Account Number is required'));
+    }
+
+    try {
+        const validUser = await User.findOne({accountNumber : id}).select("-password");
+
+        if (!validUser) {
+            return next(errorHandler(404, 'User not found'));
+        }
+
+        validUser.userType = modifyUser.userType || validUser.userType;
```

Comment:
this is too lengthy, there are update methods in mongoose for updating a single field